src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -553,7 +536,11 @@ private unsafe int HandleEventShutdownComplete(ref SHUTDOWN_COMPLETE data)
     {
         if (data.ConnectionShutdown != 0)
         {
-            Exception exception = ThrowHelper.GetConnectionAbortedException(_connectionState.AbortErrorCode);
+            // If it's local shutdown by App, i.e.: this side called QuicConnection.CloseAsync, throw OperationAbortedException.
+            Exception exception = data.ConnectionShutdownByApp != 0 && data.ConnectionClosedRemotely == 0 ?
+                ThrowHelper.GetOperationAbortedException() :
+                // TODO: this will contain 0 for transport shutdown, we should propagate transport error code.
+                ThrowHelper.GetConnectionAbortedException(data.ConnectionShutdownByApp != 0 ? (long)data.ConnectionErrorCode : 0);
```

Comment:
We should not throw with `ConnectionAborted` at all if the connection was shut down by transport. In case of shutdown by transport, we should get the right exception according to the `CloseStatus` provided in the `SHUTDOWN_INITIATED_BY_TRANSPORT` event on the connection

Reply:
We didn't have that ability before and we don't have it now.
We do have a transport error code, but we ignore it in both cases (SHUTDOWN_INITIATED_BY_TRANSPORT and here) because we haven't come up with solution where/how to put it.

Reply:
Yeah, it is something we need to follow-up on, can we put there InternalError for now? I really dislike the idea of Shutdown-by-transport being surfaced as Shutdown-by-application.

Reply:
I took the liberty of assuming IdleConnection for local transport closure, which from to quick check of msquic looks like it's the most probably cause when you're at the point you already have some streams open (I might have overlooked something and it might obviously change). And I put a comment and linked the issue there. We can solve it for .NET 8.
If you think my assumption is wrong, I'll put `InternalError` there as well.

Reply:
If we don't, mind misrepresenting actual protocol violations as idle timeout for now, then I am fine with it.

Once I am done with the task I am currently working on, I would like to take a look at the follow-up, I feel it would be better to try to squeeze it in 7.0.